io/disk/Avago_storage_adapter/avago3008.py

```diff
@@ -92,42 +118,69 @@ def set_online_offline(self, state):
         if process.system(cmd, ignore_status=True, shell=True) != 0:
             self.fail("Failed to set drive to %s" % state)
 
-    def test_setoffline(self):
+    def adapter_status(self):
 
         """
-        This function set's a drive to offline
+        This is a helper function, to check the status of the adapter
         """
-        self.set_online_offline("offline")
+        cmd = "./sas3ircu %d status" % self.controller
+        if process.system(cmd, ignore_status=True, shell=True) != 0:
+            self.fail("Failed to display the status of the adapter")
```

Comment:
@narasimhan-v  O/P is as follows:

./sas3ircu 0 status
Avago Technologies SAS3 IR Configuration Utility.
Version 13.00.00.00 (2016.03.08) 
Copyright (c) 2009-2016 Avago Technologies. All rights reserved. 

     >>>>>>  THIS IS A DEVELOPER BUILD - FOR EVALUATION ONLY  <<<<<< 

Background command progress status for controller 0...
IR Volume 1
  Volume ID                               : 323
  PI Supported                            : No
  Current operation                       : None
  Volume status                           : Enabled
  Volume state                            : Optimal
  Volume wwid                             : 08dfd8f38f210575
  Physical disk I/Os                      : Not quiesced
SAS3IRCU: Command STATUS Completed Successfully.
SAS3IRCU: Utility Completed Successfully.